Applications/deviceconfiguration/simpleconfig.py

```diff
@@ -40,14 +40,75 @@
-    print e
+except requests.exceptions.RequestException as e:
+    # Some error occurred
+    print r.text
 
+#Check to see if programming was successful (HTTP 204 response)
+if r.status_code != 204:
+    print r.text
+else:
+    # Programming apparently successful. Let unit reboot and then query for flash data
+    
+    timer = 5 #  Wait five seconds
+    print "Programmed Faraday, waiting {0} seconds for reboot".format(str(timer))
+    while(timer > 0):
+        time.sleep(1) # Sleep to allow unit to process, polling and slow, not sure why THIS slow...
+        timer += -1
+
+    try:
+        r = requests.get("http://127.0.0.1:8002", params={'callsign': str(local_device_callsign), 'nodeid': int(local_device_node_id)})
+    except requests.exceptions.RequestException as e:
+        # Some error occurred
+        print r.text
+
+    # Obtain JSON response with data from Faraday
+    raw_unit_json = r.json()
+
+    # Decode and depickle (serialize) device configuration parsed dictionary data
+    b64_unit_json = base64.b64decode(raw_unit_json['data'])
+    unit_configuration_dict = cPickle.loads(b64_unit_json)
+    
+    #Print configuration values
+    print "\n************************************"
+    
+    print "POST CONFIGURATION UPDATE"
+    print "\nBASIC"
+    print "Unit Callsign:", str(unit_configuration_dict['local_callsign'])[0:unit_configuration_dict['local_callsign_length']]
+    print "ID:", str(unit_configuration_dict['local_callsign_id'])
+    print "CONFIGBOOTBITMASK:", bin(unit_configuration_dict['configuration_bitmask'])
+    print "GPIO_P3:", bin(unit_configuration_dict['default_gpio_port_3_bitmask'])
+    print "GPIO_P4:", bin(unit_configuration_dict['default_gpio_port_4_bitmask'])
+    print "GPIO_P5:", bin(unit_configuration_dict['default_gpio_port_5_bitmask'])
+    print "\nRF"
+    print "BOOT_FREQUENCY_MHZ 0:", str(unit_configuration_dict['default_boot_freq_0'])
+    print "BOOT_FREQUENCY_MHZ 1:", str(unit_configuration_dict['default_boot_freq_1'])
+    print "BOOT_FREQUENCY_MHZ 2:", str(unit_configuration_dict['default_boot_freq_2'])
+    print "BOOT_RF_POWER:", str(unit_configuration_dict['default_rf_power'])
+    print "\nGPS"
+    print "DEFAULT_LATITUDE:", str(unit_configuration_dict['default_gps_latitude'])
+    print "DEFAULT_LATITUDE_DIR:", str(unit_configuration_dict['default_gps_latitude_dir'])
+    print "DEFAULT_LONGITUDE:", str(unit_configuration_dict['default_longitude'])
+    print "DEFAULT_LONGITUDE_DIR:", str(unit_configuration_dict['default_longitude_dir'])
+    print "DEFAULT_ALTITUDE:", str(unit_configuration_dict['default_altitude'])
+    print "DEFAULT_ALTITUDE_UNITS:", str(unit_configuration_dict['default_altitude_units'])
+    print "GPS_BOOT_BIT & GPS_PRESENT_BIT:", bin(unit_configuration_dict['gps_boot_bitmask'])
+    print "\nTELEMETRY"
+    print "UART_TELEMETRY_BOOT_BIT & RF_TELEMETRY_BOOT_BIT:", bin(unit_configuration_dict['telemetry_boot_bitmask'])
```

Comment:
Since this is a binary print of the bitmask you should show all leading zeros: http://stackoverflow.com/questions/16926130/python-convert-to-binary-and-keep-leading-zeros